DB/src/main/java/io/deephaven/db/util/DelegatedScriptSession.java

```diff
@@ -0,0 +1,158 @@
+package io.deephaven.db.util;
+
+import io.deephaven.db.tables.select.QueryScope;
+import io.deephaven.db.util.liveness.LivenessReferent;
+import io.deephaven.db.util.scripts.ScriptPathLoader;
+import io.deephaven.db.util.scripts.ScriptPathLoaderState;
+import io.deephaven.lang.parse.api.CompletionParseService;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.lang.ref.WeakReference;
+import java.nio.file.Path;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Supplier;
+
+/**
+ * The delegated script session delegates all calls to another script session. When evaluating a script
+ * it massages the Changes to the QueryScope so that any modifications that are being seen for the first
+ * time by the api-client come across as new entries and not as modified entries.
+ */
+public class DelegatedScriptSession implements ScriptSession {
```

Comment:
In what context is this necessary? I think I missed it elsewhere in the review...